### PR TITLE
Use call_iedb module to make predictions

### PR DESCRIPTION
**Pull request by susannasiebert**
This PR includes changes to the main run module and API to use the call_iedb module instead of a local NetMHC path. The parser is updated to parse and combine the output files from multiple IEDB prediction runs for multiple methods. Tests and example data are also updated.

The integration tests for this PR will fail until PR #78 gets merged.

This current implementation does not split the large fasta files into smaller chunks. We should do some stress testing to see if this runs into problems. I don't think that it would be a difficult change to implement though.

### CODE REVIEW

File: pvacseq/lib/main.py [hunk 1]
```diff
@@ -28,6 +35,11 @@ def main(args_input = sys.argv[1:]):
                         help="length of subpeptides(epitopes) to predict ; Multiple lengths can be specified using a comma-separated list. Typical epitope lengths vary between 8-11.",
                         type=lambda s:[int(epl) for epl in s.split(',')]
                         )
+    parser.add_argument("prediction_algorithms",
+                        choices=['NetMHCpan', 'NetMHC', 'SMMPMBEC', 'SMM', 'NetMHCcons', 'PickPocket'],
```
Comment (acoffman) [inline pvacseq/lib/main.py]: It seems a little odd that these are harcoded here, as well as in `prediction_method_lookup`. Is it possible to generate the list of choices here based on the keys in the dict?

File: pvacseq/lib/main.py [hunk 2]
```diff
@@ -77,40 +89,42 @@ def main(args_input = sys.argv[1:]):
...
-                      "and Epitope Length", epl
-                      )
-                netmhc_cmd = "%s -a %s -l %d %s -x %s" % (
-                    args.netmhc_path, a, epl,
+    iedb_output_files = []
+    for method in args.prediction_algorithms:
+        iedb_method = prediction_method_lookup(method)
+        for epl in args.epitope_length:
+            for a in args.allele:
+ #               if a in netmhc_output:
```
Comment (acoffman) [inline pvacseq/lib/main.py]: Looks like a stray comment got left in here?

File: pvacseq/lib/main.py [hunk 3]
```diff
@@ -77,40 +89,42 @@ def main(args_input = sys.argv[1:]):
...
-                print("NetMHC allele not valid.  Please check using:")
-                print(args.netmhc_path,"-A")
-            print("Completed")
+                    iedb_out,
+                    iedb_method,
+                    a,
+                    str(epl),
+                ])
+                iedb_output_files.append(iedb_out)
+#                else:
```
Comment (acoffman) [inline pvacseq/lib/main.py]: Looks like a few stray comments in general in this file!

File: pvacseq/lib/parse_output.py [hunk 4]
```diff
@@ -5,8 +5,18 @@
 import sys
 import os
 from math import ceil
+from statistics import median
 
-import pdb
+def prediction_method_lookup(prediction_method):
```
Comment (acoffman) [inline pvacseq/lib/parse_output.py]: Maybe this needs to be moved to a shared module of some sort as it appears here and in `main.py` as well as being used in the command line interface `choices`

File: pvacseq/lib/main.py [hunk 5]
```diff
@@ -77,40 +89,42 @@ def main(args_input = sys.argv[1:]):
...
-                    args.netmhc_path, a, epl,
+    iedb_output_files = []
+    for method in args.prediction_algorithms:
+        iedb_method = prediction_method_lookup(method)
+        for epl in args.epitope_length:
+            for a in args.allele:
+ #               if a in netmhc_output:
+                iedb_out = os.path.join(args.output_dir, ".".join([args.sample_name, a, str(epl), iedb_method, "tsv"]))
+                print("Running IEDB on Allele %s and Epitope Length %s with Method %s" % (a, epl, method))
+                lib.call_iedb.main([
```
Comment (acoffman) [inline pvacseq/lib/main.py]: If `call_iedb.main()` can `sys.exit()` if the request fails, do we need to catch that somehow? Will we leave half completed or temp files laying around if we get a failed HTTP request?

File: pvacseq/lib/main.py (hunk 2 above)
Comment (susannasiebert) [inline pvacseq/lib/main.py]: Lines 92-96 had functionality to look up the available list of alleles from NetMHC. This was then checked on this line against the list of alleles input by the user. IEDB has it's own post request for looking up available alleles. Unfortunately, that is currently broken. I put in a request to have it fixed. Once it is available we should incorporate this check again. I'm not sure if we should delete the commented out code until then.

Comment (acoffman) [inline pvacseq/lib/main.py]: Cool, I'm fine leaving it there as a reminder to fix it on our end then!

File: pvacseq/lib/parse_output.py (hunk 4 above)
Comment (susannasiebert) [inline pvacseq/lib/parse_output.py]: I just pushed a commit that moves this code into a shared module.

File: pvacseq/lib/main.py (hunk 5 above)
Comment (susannasiebert) [inline pvacseq/lib/main.py]: call_iedb doesn't write do its output file until after the HTTP request succeeds so that shouldn't be a problem.

**Comment by acoffman** [comment]
+1 🌴

**Comment by agraubert** [comment]
+1

**Pull request merged**